lib/membrane_rtc_engine/engine.ex

```diff
@@ -399,9 +480,8 @@ defmodule Membrane.RTC.Engine do
   end
 
   @impl true
-  def handle_other({:add_peer, peer_id, data}, _ctx, state) do
-    {actions, state} = do_accept_new_peer(peer_id, data, state)
-
+  def handle_other({:add_peer, peer}, _ctx, state) do
+    {actions, state} = do_accept_new_peer(peer, state)
```

Comment:
So now we assume that we will receive from custom backend logic the Peer struct?

Reply:
Yeap because I made Peer module public. It's easier to operate on whole struct than on specific fields

Reply:
But in `handle_media_event({type: :join},...)` client doesn't send Peer struct. Isn't it a little bit inconvenient, that we have two ways to add Peer but the client has to pass two different types of data? @mickel8

Reply:
Joining with Media Event `join` is something different. User doesn't call add_peer when joining using Media Event. It just accepts or denies peer passing its id. In `handle_media_event({type: join, ...})` we create a new peer struct so I think it's ok.  The question is whether we should have two differernt mechanisms allowing to join to the RTC Engine. @mat-hek suggested removing possibility to join to the RTC Engine using Media Event but I haven't thought over this yet.